src/Lexer.js

```diff
@@ -68,6 +69,10 @@ export default class Lexer implements LexerInterface {
         const start = this.pos;
         this.pos += match[0].length;
         const end = this.pos;
-        return new Token(text, start, end, this);
+        if (/%[^\n]*[\n]/.test(text)) {
```

Comment:
Is there a way to factor out this regex so we don't have to duplicate it here and above? Or maybe we could just test for a single `^%` so it doesn't have to be duplicated?